### PR TITLE
[EC-71] Add API support for neighbor cells

### PR DESCRIPTION
**Pull request by tboinske-skyhook**
embedded-client API support for neighbor cells

### CODE REVIEW

File: libel/beacons.h [hunk 1]
```diff
@@ -121,6 +125,8 @@ struct nbiot {
     uint16_t mnc;
```
Comment (sissakov-skyhook) [inline libel/beacons.h:125]: not related to NMR changes but wondering why for nbiot struct there is no mcc?

Comment (gcleary) [reply libel/beacons.h:125]: @sissakov-skyhook Not sure why github was hiding it, but if you expand all, mcc and mnc are both in the mbiot struct.

File: libel/beacons.h [hunk 2]
```diff
@@ -121,6 +125,8 @@ struct nbiot {
...
     uint32_t e_cellid;
     uint16_t tac;
+    uint16_t ncid;
+    int32_t earfcn;
```
Comment (gcleary) [inline libel/beacons.h:129]: nbiot earfcn field can be 16 bit like umts uarfcn.

File: libel/libel.c [hunk 3]
```diff
@@ -343,14 +343,16 @@ Sky_status_t sky_add_ap_beacon(Sky_ctx_t *ctx, Sky_errno_t *sky_errno, uint8_t m
  *  @param e_cellid lte beacon identifier 28bit (0-268,435,456)
  *  @param mcc mobile country code (200-799)
  *  @param mnc mobile network code (0-999)
+ *  @param pci mobile pci (0-503, -1 if unknown)
+ *  @param earfcn mobile earfcn (0-45589, -1 if unknown, -2 if unknown but same as serving cell))
```
Comment (gcleary) [inline libel/libel.c:347]: A quick google search would indicate LTE earfcn ranges between 0-65535! It's 18 bit so why 0-45589?

File: libel/libel.c [hunk 4]
```diff
@@ -446,14 +450,16 @@ Sky_status_t sky_add_cell_gsm_beacon(Sky_ctx_t *ctx, Sky_errno_t *sky_errno, uin
  *  @param ucid umts cell identifier 28bit (0-268,435,456)
  *  @param mcc mobile country code (200-799)
  *  @param mnc mobile network code  (0-999)
+ *  @param psc mobile primary scrambling code (0-511, -1 if unknown)
+ *  @param uarfcn mobile uarfcn (412-10833, -1 if unknown, -2 if unknown but same as serving cell))
```
Comment (gcleary) [inline libel/libel.c:454]: Small point, but in comment you have uarfcn (412-10833), but Wikipedia says (412-10838).

File: libel/libel.c [hunk 5]
```diff
@@ -545,14 +553,16 @@ Sky_status_t sky_add_cell_cdma_beacon(Sky_ctx_t *ctx, Sky_errno_t *sky_errno, ui
  *  @param mnc mobile network code  (0-999)
  *  @param e_cellid nbiot beacon identifier (0-268,435,456)
  *  @param tac nbiot tracking area code identifier (1-65,535), 0 if unknown
+ *  @param ncid mobile cell ID (0-503, -1 if unknown)
+ *  @param earfcn mobile earfcn (0-45589, -1 if unknown, -2 if unknown but same as serving cell))
```
Comment (gcleary) [inline libel/libel.c:557]: For nb-iot band 43 seems to have a range for earfcn of (43590-45589), however there are further bands with higher frequencies. Why are we limiting this to 0-45589?

File: libel/beacons.h (hunk 2 above)
Comment (tboinske-skyhook) [reply libel/beacons.h:129]: But the earfcn range is 0-45589, -1, -2. A 16-bit int has a range of −32768 to 32767, and therefore won't suffice.

File: libel/libel.c (hunk 3 above)
Comment (tboinske-skyhook) [reply libel/libel.c:347]: I took the range from our API server spec: https://github.com/SkyhookWireless/wps2-protocol/blob/master/xml/wps2/location.xsd

File: libel/beacons.h (hunk 2 above)
Comment (gcleary) [reply libel/beacons.h:129]: Of course you are right, https://team.skyhook.com/confluence/display/DOCS/XML++Location+API+v2.28#XMLLocationAPIv2.28-LTETower
is wrong!

File: libel/libel.c (hunk 4 above)
Comment (tboinske-skyhook) [reply libel/libel.c:454]: Same comment as above (https://github.com/SkyhookWireless/wps2-protocol/blob/c4fb3190b69fd61d42cb25ac4059d8d0925fd173/xml/wps2/location.xsd#L361). @sissakov-skyhook - can you comment? Which is correct?

File: libel/libel.c (hunk 5 above)
Comment (tboinske-skyhook) [reply libel/libel.c:557]: Same comment as above. Just going on what's in our XSD doc. @sissakov-skyhook ?

Comment (sissakov-skyhook) [reply libel/libel.c:557]: The supported frequency bands are updated constantly.  All the new bands (above 41) were introduced after Rel14 so not really deployed yet. With that being said, sooner or later we will need to update the entire system to support bands beyond 16bit even (as we just introduced for 5G where channel is 24bit). I assume we are not doing real range check on the client so OK to include any channels up to 16bit (65535) for now.

File: libel/beacons.h (hunk 2 above)
Comment (tboinske-skyhook) [reply libel/beacons.h:129]: It's the need to accommodate the special negative flag values that spoils the party.

Comment (gcleary) [reply libel/beacons.h:129]: yup, that's right.

File: libel/utilities.c [hunk 6]
```diff
@@ -1448,6 +1474,253 @@ int64_t get_gnss_age(Sky_ctx_t *ctx, uint32_t idx)
...
+    return cell->h.type;
+}
+
+/*! \brief Get cell id1
+ *
+ *  @param cell Pointer to beacon (cell)
+ *
+ *  @return cell id1, -1 if not available
+ */
+int64_t get_cell_id1(Beacon_t* cell)
```
Comment (gcleary) [inline libel/utilities.c:1511]: Suggest beacons.h should have id1..6 in comments against each cell field.

Comment (tboinske-skyhook) [reply libel/utilities.c:1511]: Yes, will do. Incidentally, an ideal future enhancement would be to replace all of the per-type identifiers (mcc, mnc, lac, tac, ucid, nid, ncid, etc.) with id1..id6 everywhere. In fact, I would be inclined to do it now if I had the time. So instead I created these mapping functions.

**Pull request merged**